# --- SDE-COPYRIGHT-NOTE-BEGIN ---
# This copyright note is auto-generated by ./scripts/Create-CopyPatch.
# 
# Filename: package/.../hal/hal-0.5.7.1-linux-header-2.6.18.patch
# Copyright (C) 2006 The OpenSDE Project
# 
# More information can be found in the files COPYING and README.
# 
# This patch file is dual-licensed. It is available under the license the
# patched project is licensed under, as long as it is an OpenSource license
# as defined at http://www.opensource.org/ (e.g. BSD, X11) or under the terms
# of the GNU General Public License as published by the Free Software
# Foundation; either version 2 of the License, or (at your option) any later
# version.
# --- SDE-COPYRIGHT-NOTE-END ---

--- ./hald/linux/probing/linux_dvd_rw_utils.c.orig	2006-09-29 23:14:00.000000000 +0200
+++ ./hald/linux/probing/linux_dvd_rw_utils.c	2006-09-29 23:26:04.000000000 +0200
@@ -149,8 +149,8 @@
 		errno = EIO;
 		ret = -1;
 		if (cmd->sg_io.masked_status & CHECK_CONDITION) {
-			CREAM_ON_ERRNO (cmd->sg_io.sbp);
-			ret = ERRCODE (cmd->sg_io.sbp);
+			CREAM_ON_ERRNO (((unsigned char *)cmd->sg_io.sbp));
+			ret = ERRCODE (((unsigned char *)cmd->sg_io.sbp));
 			if (ret == 0)
 				ret = -1;
 		}
@@ -172,7 +172,7 @@
 		cmd->sg_io.mx_sb_len = sizeof (cmd->_sense);
 		cmd->sg_io.cmdp = cmd->cgc.cmd;
 		cmd->sg_io.sbp = cmd->_sense.u;
-		cmd->sg_io.flags = SG_FLAG_LUN_INHIBIT | SG_FLAG_DIRECT_IO;
+		cmd->sg_io.flags = SG_FLAG_UNUSED_LUN_INHIBIT | SG_FLAG_DIRECT_IO;
 	}
 	cmd->sg_io.cmd_len = i + 1;
 	cmd->cgc.cmd[i] = arg;
